include/zenoh_commons.h

```diff
@@ -1967,6 +2029,47 @@ ZENOHC_API struct z_owned_reply_t z_reply_null(void);
  */
 ZENOHC_API
 struct z_sample_t z_reply_ok(const struct z_owned_reply_t *reply);
+/**
+ * The sample's attachment.
+ *
+ * `sample` is aliased by the return value.
+ */
+ZENOHC_API struct z_attachment_t z_sample_attachment(const struct z_sample_t *sample);
+/**
+ * The encoding of the payload.
+ */
+ZENOHC_API struct z_encoding_t z_sample_encoding(const struct z_sample_t *sample);
+/**
+ * The Key Expression of the sample.
+ *
+ * `sample` is aliased by its return value.
+ */
+ZENOHC_API struct z_keyexpr_t z_sample_keyexpr(const struct z_sample_t *sample);
+/**
+ * The sample's kind (put or delete).
+ */
+ZENOHC_API enum z_sample_kind_t z_sample_kind(const struct z_sample_t *sample);
+/**
+ * Returns the sample's payload after incrementing its internal reference count.
+ *
+ * Note that other samples may have received the same buffer, meaning that mutating this buffer may
+ * affect the samples received by other subscribers.
+ */
+ZENOHC_API struct z_owned_buffer_t z_sample_owned_payload(const struct z_sample_t *sample);
+/**
+ * The sample's data, the return value aliases the sample.
+ *
+ * If you need ownership of the buffer, you may use `z_sample_owned_payload`.
+ */
+ZENOHC_API struct z_bytes_t z_sample_payload(const struct z_sample_t *sample);
```

Comment:
Todo: replace this with one method
`ZENOHC_API struct z_payload_t z_sample_payload(const struct z_sample_t *sample);`